Build failure: `mame` on `*-darwin` (missing `sdl3`)

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

```console
$ # On macOS, either:
$ nix-build '<nixpkgs>' -A mame
$ # or:
$ nix build 'nixpkgs#mame'
```

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

[`x86_64`](https://hydra.nixos.org/build/322988757), [`aarch64`](https://hydra.nixos.org/build/322988759)

### Relevant log output

```console
[…]
sh: sw_vers: command not found
Package sdl3 was not found in the pkg-config search path.
Perhaps you should add the directory containing `sdl3.pc'
to the PKG_CONFIG_PATH environment variable
No package 'sdl3' found
Package sdl3 was not found in the pkg-config search path.
Perhaps you should add the directory containing `sdl3.pc'
to the PKG_CONFIG_PATH environment variable
No package 'sdl3' found
Package sdl3 was not found in the pkg-config search path.
Perhaps you should add the directory containing `sdl3.pc'
to the PKG_CONFIG_PATH environment variable
No package 'sdl3' found
Package sdl3 was not found in the pkg-config search path.
Perhaps you should add the directory containing `sdl3.pc'
to the PKG_CONFIG_PATH environment variable
No package 'sdl3' found
Package sdl3 was not found in the pkg-config search path.
Perhaps you should add the directory containing `sdl3.pc'
to the PKG_CONFIG_PATH environment variable
No package 'sdl3' found
[…]
Compiling src/osd/modules/input/input_common.cpp...
Compiling src/devices/bus/a800/sic.cpp...
Compiling src/devices/bus/a800/sparta.cpp...
../../../../../src/osd/modules/input/input_common.cpp:32:10: fatal error: 'SDL3/SDL.h' file not found
   32 | #include <SDL3/SDL.h>
      |          ^~~~~~~~~~~~
1 error generated.
make[2]: *** [osd_sdl3.make:1292: ../../../../osx_clang/obj/x64/Release/osd_sdl3/src/osd/modules/input/input_common.o] Error 1
make[1]: *** [Makefile:19: osd_sdl3] Error 2
make[1]: *** Waiting for unfinished jobs....
[…]
```

### Additional context

The most recent version of MAME ([0.286](https://github.com/mamedev/mame/releases/tag/mame0286), which was added to Nixpkgs in #494243) adds support for SDL3. For some reason, the derivation seems to end up trying to build the SDL3 variant of MAME instead of the SDL2 variant, even though `buildInputs` still only contains `SDL2` (a.k.a. `sdl2-compat`), and fails because it can't find `SDL3/SDL.h`.

The Linux builds ([`x86_64`](https://hydra.nixos.org/build/322988761), [`aarch64`](https://hydra.nixos.org/build/322988763)) seem to end up building the SDL3 version of MAME as well, but are actually succeeding. I guess enough of `sdl3` is making it into the path through `sdl2-compat` to make it work. Not sure why that's different between the two platforms.

### System metadata

 - system: `"x86_64-darwin"`
 - host os: `Darwin 24.5.0, macOS 15.5`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.31.3`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/store/5xgk3f6y9b8nlq99v3p18yfs11xjjjq6-nixpkgs/nixpkgs`
   - Note: the channel above isn't on the failing version yet, but I can reproduce when manually checking out #494243 or later, or using the flake.

### Notify maintainers

`mame` maintainers: @thiagokokada @DimitarNestorov

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+label%3A%220.kind%3A+build+failure%22).
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/ma/mame/package.nix
+++ b/pkgs/by-name/ma/mame/package.nix
@@ -3,8 +3,8 @@
   stdenv,
   fetchFromGitHub,
   alsa-lib,
-  SDL2,
-  SDL2_ttf,
+  sdl3,
+  sdl3-ttf,
   copyDesktopItems,
   expat,
   fetchurl,
@@ -55,6 +55,7 @@ stdenv.mkDerivation (finalAttrs: {
     "CC=${stdenv.cc.targetPrefix}cc"
     "CXX=${stdenv.cc.targetPrefix}c++"
     "TOOLS=1"
+    "OSD=sdl3"
     "USE_LIBSDL=1"
     # "USE_SYSTEM_LIB_ASIO=1"
     "USE_SYSTEM_LIB_EXPAT=1"
@@ -84,8 +85,8 @@ stdenv.mkDerivation (finalAttrs: {
     rapidjson
     pugixml
     glm
-    SDL2
-    SDL2_ttf
+    sdl3
+    sdl3-ttf
     sqlite
     libsForQt5.qtbase
   ]
@@ -135,11 +136,9 @@ stdenv.mkDerivation (finalAttrs: {
   # This replaces the `sw_vers` call with the macOS version actually being
   # targeted, so everything gets linked correctly.
   + lib.optionalString stdenv.hostPlatform.isDarwin ''
-    for file in scripts/src/osd/{mac,sdl}.lua; do
-      substituteInPlace "$file" --replace-fail \
-        'backtick("sw_vers -productVersion")' \
-        "os.getenv('MACOSX_DEPLOYMENT_TARGET') or '$darwinMinVersion'"
-      done
+    substituteInPlace scripts/src/osd/sdl3.lua --replace-fail \
+      'backtick("sw_vers -productVersion")' \
+      "os.getenv('MACOSX_DEPLOYMENT_TARGET') or '$darwinMinVersion'"
   '';
 
   desktopItems = [

